### PR TITLE
[Enhancement] Add more logs for GC tasks

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: be/src/storage/lake/tablet_manager.cpp [hunk 1]
```diff
@@ -814,8 +815,16 @@ void* metadata_gc_trigger(void* arg) {
         for (const auto& root : roots) {
             // TODO: limit GC concurrency
             st = thread_pool->submit_func([=]() {
+                auto t1 = std::chrono::steady_clock::now();
                 auto r = metadata_gc(root, tablet_mgr, master_info.min_active_txn_id);
-                LOG_IF(WARNING, !r.ok()) << "Fail to do metadata gc in " << root << ": " << r;
+                auto t2 = std::chrono::steady_clock::now();
+                auto cost = std::chrono::duration_cast<std::chrono::milliseconds>(t2 - t1).count();
+                if (r.ok()) {
+                    LOG(INFO) << "Finished garbage collection of metadata for directory " << root << ". cost:" << cost;
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.cpp:823]:
```suggestion
                    LOG(INFO) << "Finished garbage collection of metadata for directory " << root << ". cost:" << cost << "ms";
```

File: be/src/storage/rowset/rowset_writer.cpp [hunk 2]
```diff
@@ -203,7 +203,10 @@ Status RowsetWriter::flush_segment(const SegmentPB& segment_pb, butil::IOBuf& da
         // 1. create segment file
         auto path = Rowset::segment_file_path(_context.rowset_path_prefix, _context.rowset_id, segment_pb.segment_id());
         // use MUST_CREATE make sure atomic
-        ASSIGN_OR_RETURN(auto wfile, _fs->new_writable_file(path));
+        WritableFileOptions opts{.sync_on_close = config::sync_tablet_meta,
+                                 .skip_fill_local_cache = false,
```
Comment (wyb) [inline be/src/storage/rowset/rowset_writer.cpp:207]: skip_fill_local_cache is true for local tablet?

Comment (sduzh) [reply be/src/storage/rowset/rowset_writer.cpp:207]: This is unintended change, should be reverted.

**Review by wyb — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**